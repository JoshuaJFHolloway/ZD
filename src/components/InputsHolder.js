import React from 'react';
import Header from './Header';
import InputTitleStyle from './styledComponents/inputTitle';
import ButtonStyle from './styledComponents/Button';
import InputStyle from './styledComponents/Input';

const InputsHolder = (props) => {
  return (
    <div>
      <InputTitleStyle>
        <Header
          title={props.title}
        />
      </InputTitleStyle>

      <InputStyle
        type={props.type}
        name={props.inputName}
        onChange={props.handleChange}
      />

      <ButtonStyle
        name={props.buttonName}
        onClick={props.handleClick}
      >
        {props.buttonName}
      </ButtonStyle>
    </div>
  )
};

export default InputsHolder;