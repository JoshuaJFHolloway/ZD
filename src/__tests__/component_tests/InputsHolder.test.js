import React from 'react';
import { shallow } from 'enzyme';
import InputsHolder from './../../components/InputsHolder';

const inputsHolder = shallow(<InputsHolder />);

describe('InputsHolder', () => {

  it('renders correctly', () => {
    expect(inputsHolder).toMatchSnapshot();
  });

  describe('rendering components', () => {

    it('has Header', () => {
      expect(inputsHolder.find('Header'));
    });

    it('has InputStyle', () => {
      expect(inputsHolder.find('InputStyle'));
    });

    it('has ButtonStyle', () => {
      expect(inputsHolder.find('ResultStyle'));
    });
  });
});