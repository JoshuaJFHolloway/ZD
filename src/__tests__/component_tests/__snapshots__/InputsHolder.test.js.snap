// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputsHolder renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputsHolder />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.h3>
          <Header
                    title={undefined}
          />
</styled.h3>,
        <styled.input
          name={undefined}
          onChange={undefined}
          type={undefined}
/>,
        <styled.button
          name={undefined}
          onClick={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Header
            title={undefined}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": undefined,
          "onChange": undefined,
          "type": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "name": undefined,
          "onClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.h3>
            <Header
                        title={undefined}
            />
</styled.h3>,
          <styled.input
            name={undefined}
            onChange={undefined}
            type={undefined}
/>,
          <styled.button
            name={undefined}
            onClick={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Header
              title={undefined}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": undefined,
            "onChange": undefined,
            "type": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "name": undefined,
            "onClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
